api/v1alpha2/bundledeployment_types.go

```diff
@@ -69,6 +69,26 @@ type BundleDeploymentSpec struct {
 	//
 	// config is provisioner specific configurations
 	Config runtime.RawExtension `json:"config,omitempty"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the configuration of preflight checks.
+	Preflight *PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the configuration for the preflight checks.
+type PreflightConfig struct {
+	//+kubebuilder:Optional
+	// CRDUpgradeSafety holds necessary configuration for the CRD Upgrade Safety preflight checks.
+	CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is the configuration for CRD upgrade safety preflight check.
+type CRDUpgradeSafetyPreflightConfig struct {
+	//+kubebuilder:default:=true
+	//+kubebuilder:Optional
```

Comment:
Let's make this required. Since the parent struct is optional, it would only make sense to specify the parent struct if you intended to explicitly specify something in the child struct.

Looking at this more closely, I think this rule should apply all the way up the stack.

So `spec.preflight` is optional, but if you specify it, the only choice right now is `spec.preflight.crdUpgradeSafety` (so let's make that required for now as well).

WDYT?

Reply:
But let's leave `CRDUpgradeSafety *CRDUpgradeSafetyPreflightConfig` as a pointer since the expectation is that we may add configurations for other preflight checks in the future.

Reply:
I totally agree with this. It doesn't make sense to specify the parent struct without explicitly specifying something in the child struct. It would make it more semantically correct to apply this rule all the way up the stack.

Reply:
Updated in [latest commit](https://github.com/operator-framework/rukpak/pull/902/commits/c7eb026d9d8bbbe3f06a830fe91ce7c0d5bfc4e4)